iopipe/iopipe.py

```diff
@@ -40,38 +34,43 @@ def log(self, key, value):
             event['n'] = value
         else:
             event['s'] = str(value)
+
         self.report.custom_metrics.append(event)
 
-    def err(self, err):
-        self.report.retain_err(err)
-        self.report.send()
-        raise err
+    def error(self, error):
+        if self.report is None:
+            warnings.warn('An exception occurred before function was decorated with IOpipe. '
+                          'This exceptionwill not be recorded.')
```

Comment:
This exception => This lambda? This invocation?

Reply:
In this case it is if `error()` is called before the function is decorated or if it is called outside the scope of a decorated function being called. Not sure on the language as this could be outside an actual invocation.

Also, do we want to continue to support this method? We catch exceptions within the decorator itself. Is there a case where a dev would want to record an exception explicitly? Since that effectively sends a report, and this can only happen once per invocation, it seems like just raising the exception within the function itself is what the user would want from an IOpipe standpoint. I think we've already deprecated this method in the nodejs agent, should we deprecate this as well?

Reply:
➕ on deprecating it here, it's a difference between the interfaces. Let's do that.
Thanks for the clarification, then the typo is only missing the space: "this exceptionwill" => "this exception will" [maybe exception => error]